e2e/test/helpers/TestDevice.cs

```diff
@@ -52,25 +48,14 @@ private TestDevice(Device device, IAuthenticationMethod authenticationMethod)
         /// <param name="type">The way the device will authenticate</param>
         public static async Task<TestDevice> GetTestDeviceAsync(string namePrefix, TestDeviceType type = TestDeviceType.Sasl)
         {
-            string prefix = namePrefix + type + "_";
-
-            try
-            {
-                await s_semaphore.WaitAsync().ConfigureAwait(false);
```

Comment:
Why do we have a semaphore here??

Reply:
The service client operations needed to create a new test device should already be thread safe, so I agree with removing this semaphore

Reply:
It's just an HTTP request, after all